src/CMakeLists.txt

```diff
@@ -0,0 +1,79 @@
+add_subdirectory(vmec_interface)
+
+
+add_executable(stella.x  ${SOURCES} )
+
+#set flags here
+#TODO add the most common flags needed for stella
+
+set(dialect "-ffree-line-length-none -fno-backslash -I /usr/include")
+set(bounds "-fbounds-check")
+set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} ${dialect} -march=native")
+set(CMAKE_Fortran_FLAGS_DEBUG "${CMAKE_Fortran_FLAGS_DEBUG} ${bounds} -pedantic -g -Wall -fimplicit-none")
+if(PRECISION STREQUAL "DOUBLE")
+	set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -fdefault-real-8 -fdefault-double-8")
+endif()
```

Comment:
It's not good to touch the `CMAKE_<lang>_FLAGS` variables, these should really be left for the user to set.
These flags are also specific to gfortran, and won't work with Intel, for example.

Here's how we set the precision flags in another project:

```cmake
# This option is best put in the top-level CMakeLists.txt, near the top
option(GK_ENABLE_DOUBLE "Promotes precisions of real and complex to double" ON)

if (GK_ENABLE_DOUBLE)
  target_compile_options(gkutils PUBLIC
    $<$<Fortran_COMPILER_ID:GNU>:
      -fdefault-real-8 -fdefault-double-8 >
    $<$<Fortran_COMPILER_ID:Intel>:
      -r8 >
    )
endif()
```

and some debug flags:

```cmake
# Add some extra flags for Debug configs

target_compile_options(gkutils PRIVATE
  $<$<CONFIG:Debug>:
    $<$<Fortran_COMPILER_ID:GNU>:
      -g -Wall -fimplicit-none -fbounds-check >
    $<$<Fortran_COMPILER_ID:Intel>:
      -g -implicitnone -warn all -nogen-interfaces -CB -traceback >
    >
  )
```

The horrible-looking `$<...>` bits are [generator expressions](https://cmake.org/cmake/help/latest/manual/cmake-generator-expressions.7.html), and are very useful for doing this sort of thing, where we set flags depending on the compiler